api/org/commcare/api/persistence/SqliteIndexedStorageUtility.java

```diff
@@ -0,0 +1,460 @@
+                    org.commcare.modern.database.DatabaseHelper.DATA_COL + " FROM " + this.tableName + ";";
+            preparedStatement = connection.prepareStatement(sqlQuery);
+            resultSet = preparedStatement.executeQuery();
+        } catch (SQLException | ClassNotFoundException e) {
+            e.printStackTrace();
+        }
+        return new JdbcSqlStorageIterator<T>(preparedStatement, resultSet, this.getNumRecords(), this);
+    }
+
+    @Override
+    public int getTotalSize() {
+        //serialize and test blah blah.
+        return 0;
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return this.getNumRecords() <= 0;
+    }
+
+    @Override
+    public byte[] readBytes(int id) {
+        PreparedStatement preparedStatement = null;
+        Connection c = null;
+        try {
+            c = getConnection();
+            preparedStatement = SqlHelper.prepareIdSelectStatement(c, this.tableName, id);
+            if (preparedStatement == null) {
+                return null;
+            }
+            ResultSet rs = preparedStatement.executeQuery();
+            return rs.getBytes(org.commcare.modern.database.DatabaseHelper.DATA_COL);
+        } catch (Exception e) {
+            System.out.println("SqliteIndexedStorageUtility readBytes exception: " + e);
+        } finally{
+            try {
+                if (preparedStatement != null) {
+                    preparedStatement.close();
+                }
+                if (c != null) {
+                    c.close();
+                }
+            } catch (SQLException e) {
+                e.printStackTrace();
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public void update(int id, Persistable p) throws StorageFullException {
+        Connection c = null;
+        try {
+            c = getConnection();
+            SqlHelper.updateToTable(c, tableName, p, id);
+            c.close();
+        } catch (SQLException | ClassNotFoundException e) {
+            e.printStackTrace();
+        } finally{
+            try {
+                if (c != null) {
+                    c.close();
+                }
+            } catch (SQLException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    @Override
+    public void remove(int id) {
+        Connection c = null;
+        try {
+            c = getConnection();
+            SqlHelper.deleteIdFromTable(c, tableName, id);
+            c.close();
+        } catch (SQLException | ClassNotFoundException e) {
+            throw new RuntimeException(e);
+        } finally{
+            try {
+                if (c != null) {
+                    c.close();
+                }
+            } catch (SQLException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    @Override
+    public void remove(Persistable p) {
+        this.remove(p.getID());
```

Comment:
Not a huge deal given the simplicity of this method, but it isn't covered by the test suite